### PR TITLE
feat: backport flatCallTracer

### PR DESCRIPTION
**Pull request by draganm**
## 1. Purpose or design rationale of this PR

Backport of `flatCallTracer` added in Geth 1.11.3

## 2. PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [x] feat: A new feature
- [ ] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

## 3. Deployment tag versioning

Has the version in `params/version.go` been updated?

- [ ] This PR doesn't involve a new deployment, git tag, docker image tag, and it doesn't affect traces
- [ ] Yes

## 4. Breaking change label

Does this PR have the `breaking-change` label?

- [x] This PR is not a breaking change
- [ ] Yes

### CODE REVIEW

File: eth/tracers/native/call_flat.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+func flatFromNested(input *callFrame, traceAddress []int, convertErrs bool, ctx *tracers.Context) (output []flatCallFrame, err error) {
+	var frame *flatCallFrame
+	switch input.Type {
+	case vm.CREATE.String(), vm.CREATE2.String():
+		frame = newFlatCreate(input)
+	case vm.SELFDESTRUCT.String():
+		frame = newFlatSuicide(input)
+	case vm.CALL.String(), vm.STATICCALL.String(), vm.CALLCODE.String(), vm.DELEGATECALL.String():
+		frame = newFlatCall(input)
```
Comment (0xmountaintop) [inline eth/tracers/native/call_flat.go]:
```suggestion
	switch vm.StringToOp(input.Type) {
	case vm.CREATE, vm.CREATE2:
		frame = newFlatCreate(input)
	case vm.SELFDESTRUCT:
		frame = newFlatSuicide(input)
	case vm.CALL, vm.STATICCALL, vm.CALLCODE, vm.DELEGATECALL:
		frame = newFlatCall(input)
```

**Review by 0xmountaintop — DISMISSED** [review]
have compared with flat call tracer on upstream
overall looks good, just some nitpicks

File: eth/tracers/native/4byte.go [hunk 2]
```diff
@@ -37,14 +37,15 @@ func init() {
...
 // Example:
-//   > debug.traceTransaction( "0x214e597e35da083692f5386141e69f47e973b2c56e7a8073b1ea08fd7571e9de", {tracer: "4byteTracer"})
-//   {
-//     0x27dc297e-128: 1,
-//     0x38cc4831-0: 2,
-//     0x524f3889-96: 1,
-//     0xadf59f99-288: 1,
-//     0xc281d19e-0: 1
-//   }
+//
```
Comment (omerfirmak) [inline eth/tracers/native/4byte.go]: Unrelated, can we revert this?

**Comment by MRabenda** [comment]
@draganm would you mind also addressing @omerfirmak comments?

File: eth/tracers/native/4byte.go (hunk 2 above)
Comment (achmand) [reply eth/tracers/native/4byte.go]: done

Comment (achmand) [reply eth/tracers/native/4byte.go]: rebased and reverted; let me know if there are any other suggestions

**Comment by MRabenda** [comment]
@HAOYUatHZ @omerfirmak anything else required to merge this PR?

**Review by 0xmountaintop — APPROVED** [review]

**Review by omerfirmak — APPROVED** [review]

**Pull request merged**